pkg/cmd/cli/cmd/login.go

```diff
@@ -178,6 +183,43 @@ func (o LoginOptions) Validate(args []string, serverFlag string) error {
 	return nil
 }
 
+func (o LoginOptions) checkConfigIsWritable(f *osclientcmd.Factory) (bool, error) {
+	kubeconfig := o.PathOptions.GetDefaultFilename()
+
+	file, err := os.OpenFile(kubeconfig, os.O_CREATE|os.O_WRONLY, 0600)
+	if err == nil {
+		file.Close()
+		return true, nil
+	}
+
+	// Whole directory does not exist and we cannot create kubeconfig file.
+	if os.IsNotExist(err) {
+		return true, nil
+	}
+
+	errmsg := kubeconfig
+
+	if os.IsPermission(err) {
+		errmsg += ": The kubeconfig file is not writable. Specify a writeable path with --config or by setting $KUBECONFIG."
+
+		if me, err := getCurrentUser(f); err == nil {
+			if me.Name == "system:admin" {
+				errmsg += " Please do not use system configuration file. This can lead to serious errors."
+			}
+		}
+	}
+
+	return false, errors.New(errmsg)
```

Comment:
Yes. I know. We should ignore them because our pre-check is not equivalent to config file creation and can throw error in the case where writing is possible.